### ISSUE TITLE
MCA crashes with high dimensions

### ISSUE THREAD

**Issue by Jovonni** [MaxHalford/prince#25]
What was the largest amount of dimensions you have tested this with?

Thanks, and kudos

**Comment by lauritk** [comment]
It depends on your memory size?  You can try to increase it with swap space.

**Comment by ttpostle** [comment]
@Jovonni 
Can you provide information about your dataset? What are its dimensions?

**Comment by MaxHalford** [comment]
Any news on this @Jovonni?

**Comment by mlagala** [comment]
Same problem here:
I have executed mca as follows:

```
mca = prince.MCA(copy=False)
mca.fit(df)
```


and i obtained:


```
---------------------------------------------------------------------------
MemoryError                               Traceback (most recent call last)
<ipython-input-38-9be1a453865b> in <module>()
----> 1 mca.fit(categorical_df[['fascia','fasc_reddito_familiare']])

/opt/conda/lib/python3.6/site-packages/prince/mca.py in fit(self, X, y)
     25 
     26         # Apply CA to the indicator matrix
---> 27         super().fit(self.one_hot_.transform(X))
     28 
     29         # Compute the total inertia

/opt/conda/lib/python3.6/site-packages/prince/ca.py in fit(self, X, y)
     60 
     61         # Compute total inertia
---> 62         self.total_inertia_ = (S @ S.T).trace()
     63 
     64         return self

MemoryError: 
```


the dataframe I'm using has 120k rows, and the code is running on a machine with 90gb of free memory + swap memory available. The crash happens a couple of second after launching the commands, and using htop it does not seems to use the swap.

I tried with different number of categorical variables from 2 to 8 variables, and each variable have at most 16 different values.

**Comment by mlagala** [comment]
@lauritk I have done the math and a dense matrix of that dimension should occupy 120 gb more or less, more than I have. So this should address the issue. Thank you.

**Comment by Arakkun** [comment]
Since cathegorial data with one-hot ecoding become a sparse matrix, it's probably better using sparse matrix for the analyses

**Comment by MaxHalford** [comment]
Indeed Prince should be able to profit from sparse encoding. Also I want to integrate scikit-learn's new `OneHotEncoder` that was released with version `0.20`. I'll get working on this soon.

**Comment by MaxHalford** [comment]
Okay so I've updated `MCA` so that it uses a one-hot encoding scheme. I've also changed `CA` so that it uses sparse operations when possible. I haven't done any benchmarks (yet) but my guess is that `MCA` will consume much less memory from now on.

This is available in master, I'll push a new version to PyPI soon.
